src/core/include/units/quantity_spec.h

```diff
@@ -233,19 +197,27 @@ struct derived_quantity_spec : detail::expr_fractions<derived_quantity_spec<>, Q
 
 #ifdef __cpp_explicit_this_parameter
   template<typename Self, Unit U>
-  [[nodiscard]] constexpr auto operator[](this const Self, U u)
+  [[nodiscard]] consteval Reference auto operator[](this const Self, U u)
     requires(dimension == detail::get_dimension_for(u))
   {
     return reference<Self{}, u>{};
   }
 #else
+  // TODO can we somehow return an explicit reference type here?
   template<Unit U>
-  [[nodiscard]] constexpr auto operator[](U u) const
+  [[nodiscard]] consteval Reference auto operator[](U u) const
     requires(dimension == detail::get_dimension_for(u))
   {
     return reference<derived_quantity_spec{}, u>{};
   }
 #endif
+
+  template<RepresentationOf<character> Rep, Unit U>
+  [[nodiscard]] consteval Quantity auto operator()(Rep&& v, U u) const
```

Comment:
```suggestion
  [[nodiscard]] constexpr Quantity auto operator()(Rep&& v, U u) const
```

Reply:
Does it have to be `constexpr` if it only forwards the work to another `constexpr` function that actually creates a quantity? All the examples were compiled correctly with `consteval`.

Reply:
If it's `consteval`, the `Rep` can't be a runtime value.

Reply:
Right, I thought I had an example with a function argument, but apparently, I did not refactor this case. Thanks!